### DISCUSSION TITLE
Реализовать универсальный дебаунсер (surf_util или отдельный пакет)

### DISCUSSION THREAD

**Discussion by saturov** [surfstudio/SurfGear#163] [Ideas]
Нужно вынести в пакет дебаунсер, который можно будет использовать для задержки событий.

- Скорее штука подходит для surf_util (или для отдельного пакета?)
- Нужно сделать настраиваемый период дебаунса

Прототип:
```dart
import 'dart:async';

const int _defaultDebounceMs = 300;

class Debouncer {
  Timer _timer;
  final int debounceMs;

  Debouncer({
    this.debounceMs = _defaultDebounceMs,
  });

  void debounce(void Function() callback) {
    if (_timer?.isActive ?? false) {
      _timer.cancel();
    }
    _timer = Timer(
      Duration(milliseconds: debounceMs),
      callback,
    );
  }

  void dispose() {
    _timer?.cancel();
  }
}
```

**Comment by HazarD2311** [comment]
extension на Stream сразу бы в виде StreamTransformer<T , T>

**Reply by saturov** [reply]
А <T, T> - это типизация чего?

**Reply by HazarD2311** [reply]
это типизицаия трансформера. Нам нужно внедриться в поток, но оставить старый тип.
Поэтому и <T,T> (а не <T,R>)
